picard/album.py

```diff
@@ -500,16 +505,24 @@ def build_tracks_cache():
                 if track:
                     yield (file, track)
                     continue
+
             # try to match by similarity
-            # TODO: find a way to speed up this part, it needs to iterate all
-            # tracks for each file, and metadata.compare() is rather complex
```

Comment:
The todo comment still applies for the new implementation. The complexity is now somewhat hidden in the calls to `candidates()` and `find_best_match()`, but the basic algorithm of iterating over all tracks, applying `metadata.compare()` on each and finally use the one with highest score stays unchanged. To be honest I don't know what we can do against that other then make sure `metadata.compare()` is as fast as it can be, though.

Reply:
Yes, but i concluded the same thing: we can't avoid to iterate over each file / track so the TODO doesn't make much sense, so i prefer to remove it.